### PR TITLE
Enable cd

### PR DESCRIPTION
**Pull request by batmat**
This PR enables automated releases. 
See https://www.jenkins.io/doc/developer/publishing/releasing-cd/ for related documentation.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
         <version>4.33</version>
     </parent>
     <artifactId>antisamy-markup-formatter</artifactId>
-    <version>2.8-SNAPSHOT</version>
+    <version>${revision}${changelist}</version>
```
Comment (timja) [inline pom.xml:9]: This doesn’t look right, normally just use changelist

Comment (jglick) [reply pom.xml:9]:
```suggestion
    <version>${changelist}</version>
```

File: pom.xml [hunk 2]
```diff
@@ -18,13 +18,16 @@
...
-      <tag>HEAD</tag>
+        <connection>scm:git:git://github.com/${gitHubRepo}.git</connection>
+        <developerConnection>scm:git:git@github.com:${gitHubRepo}.git</developerConnection>
+        <url>https://github.com/${gitHubRepo}</url>
+      <tag>${scmTag}</tag>
   </scm>
 
     <properties>
+        <revision>3.0</revision>
+        <changelist>-SNAPSHOT</changelist>
```
Comment (jglick) [inline pom.xml:29]:
```suggestion
        <changelist>999999-SNAPSHOT</changelist>
```

File: .mvn/maven.config [hunk 3]
```diff
@@ -0,0 +1,2 @@
+-Pconsume-incrementals
+-Pmight-produce-incrementals
```
Comment (jglick) [inline .mvn/maven.config:2]:
```suggestion
-Pmight-produce-incrementals
-Dchangelist.format=%d.v%s
```

**Review by jglick — CHANGES_REQUESTED** [review]
https://www.jenkins.io/doc/developer/publishing/releasing-cd/#pom-file-modifications

https://github.com/jenkins-infra/jenkins-maven-cd-action/issues/15

**Comment by jglick** [comment]
https://github.com/jenkins-infra/repository-permissions-updater/pull/2313 has been applied, good.

File: pom.xml (hunk 1 above)
Comment (batmat) [reply pom.xml]: Applied.
Indeed I simply ran `mvn incrementals:incrementalify` but then got a bit confused by https://www.jenkins.io/doc/developer/publishing/releasing-cd/#pom-file-modifications I remember.

Would it be doable to have something akin to `mvn incrementals:enable-cd` or so to do this, you think?

**Review by timja — APPROVED** [review]

**Comment by jglick** [comment]
> I simply ran mvn incrementals:incrementalify but then got a bit confused by https://www.jenkins.io/doc/developer/publishing/releasing-cd/#pom-file-modifications I remember.

Something we could do to make it clearer in docs that incrementalification is a _prerequisite_, on top of which you need to apply the described changes?

> Would it be doable to have something akin to mvn incrementals:enable-cd or so to do this, you think?

Perhaps.

File: pom.xml [hunk 4]
```diff
@@ -18,13 +18,15 @@
...
     </licenses>
     <scm>
-        <connection>scm:git:git://github.com/jenkinsci/${project.artifactId}-plugin.git</connection>
-        <developerConnection>scm:git:git@github.com:jenkinsci/${project.artifactId}-plugin.git</developerConnection>
-        <url>https://github.com/jenkinsci/${project.artifactId}-plugin</url>
-      <tag>HEAD</tag>
+        <connection>scm:git:git://github.com/${gitHubRepo}.git</connection>
+        <developerConnection>scm:git:git@github.com:${gitHubRepo}.git</developerConnection>
+        <url>https://github.com/${gitHubRepo}</url>
+      <tag>${scmTag}</tag>
```
Comment (jglick) [inline pom.xml:24]:
```suggestion
        <tag>${scmTag}</tag>
```

Crappy indentation of XML (also `<xxx/>` ⇒ `<xxx />`) is a longstanding bug of the round-trip library bundled in `maven-release-plugin` used for the two junk commits per release, which would be superseded by this PR.

**Comment by timja** [comment]
> Something we could do to make it clearer in docs that incrementalification is a _prerequisite_, on top of which you need to apply the described changes?

https://www.jenkins.io/doc/developer/publishing/releasing-cd/#prerequisites ?

**Review by jglick — APPROVED** [review]

**Pull request merged**